src/read/macho/dyld_cache.rs

```diff
@@ -28,7 +29,366 @@ where
+            } => loop {
+                if let Some(reloc) = iter.as_mut().and_then(|iter| iter.next()) {
+                    return Some(reloc);
+                }
+
+                match slide {
+                    DyldCacheSlideInfoSlice::V5(slide, page_starts) => {
+                        if *page_index < slide.page_starts_count.get(*endian).into() {
+                            let page_start: u16 =
+                                page_starts[*page_index as usize].get(*endian) as u16;
+
+                            if page_start != macho::DYLD_CACHE_SLIDE_V5_PAGE_ATTR_NO_REBASE {
+                                *iter = Some(DyldCacheRelocationPageIterator::V5 {
+                                    data: *data,
+                                    endian: *endian,
+                                    info: *info,
+                                    slide: *slide,
+                                    page_index: *page_index,
+                                    page_offset: Some(page_start.into()),
+                                });
+                            } else {
+                                *iter = None;
+                            }
+
+                            *page_index += 1;
+                        } else {
+                            return None;
+                        }
+                    }
+                }
+            },
+        }
+    }
+}
+
+/// A versioned iterator over relocations in a page
+#[derive(Debug)]
+pub enum DyldCacheRelocationPageIterator<'data, E = Endianness, R = &'data [u8]>
+where
+    E: Endian,
+    R: ReadRef<'data>,
+{
+    /// Corresponds to struct dyld_cache_slide_info5 from dyld_cache_format.h.
+    V5 {
+        /// The mapping data
+        data: R,
+        /// Endian
+        endian: E,
+        /// The mapping information
+        info: &'data macho::DyldCacheMappingAndSlideInfo<E>,
+        /// The mapping slide information
+        slide: &'data macho::DyldCacheSlideInfo5<E>,
+        /// Mapping page index
+        page_index: u64,
+        /// The current offset into the page
+        page_offset: Option<u64>,
+    },
+}
+
+impl<'data, E, R> Iterator for DyldCacheRelocationPageIterator<'data, E, R>
+where
+    E: Endian,
+    R: ReadRef<'data>,
+{
+    type Item = DyldRelocation;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        match self {
+            Self::V5 {
+                data,
+                endian,
+                info,
+                slide,
+                page_index,
+                page_offset,
+            } => {
+                if let Some(offset) = *page_offset {
+                    let mapping_offset: u64 = *page_index * slide.page_size.get(*endian) as u64;
+                    let file_offset: u64 = info.file_offset.get(*endian) + mapping_offset + offset;
+                    let pointer: macho::DyldCacheSlidePointer5 = data
+                        .read_at::<U64<E>>(file_offset)
+                        .unwrap()
```

Comment:
Don't use `unwrap`. This crate must never panic for bad input data.

Reply:
I've removed the use of `unwrap`, but noticed there is one stray in `DyldCache::parse` that might have been missed previously.